config/main.py

```diff
@@ -3227,22 +3227,48 @@ def add_route(ctx, command_str):
     config_db = ctx.obj['config_db']
     key, route = cli_sroute_to_config(ctx, command_str)
 
-    # If defined intf name, check if it exists
+    # If defined intf name, check if it belongs to interface
     if 'ifname' in route:
-        if (not route['ifname'] in config_db.get_keys('VLAN') and
+        if (not route['ifname'] in config_db.get_keys('VLAN_INTERFACE') and
             not route['ifname'] in config_db.get_keys('INTERFACE') and
-            not route['ifname'] in config_db.get_keys('PORTCHANNEL')):
+            not route['ifname'] in config_db.get_keys('PORTCHANNEL_INTERFACE')):
             ctx.fail('interface {} doesn`t exist'.format(route['ifname']))
 
+    # Alignment in case the command contains several nexthop ip
+    nh_counter = len(route['nexthop'].split(','))
```

Comment:
Is this assuming that `nexthop` is given by the command? If I use something like `config route add prefix 1.2.3.4/32 dev Ethernet8`, would it do `nh_count = 0`?

Reply:
Updated, in ```config route add prefix 1.2.3.4/32 nexthop dev Ethernet8``` case check once, in case with ip check all nexthop ip.